docs/authentication.md

```diff
@@ -0,0 +1,318 @@
+
+export default async (req, res) => {
+  if (req.method !== 'POST') return res.status(405).end();
+
+  // Exchange the DID from Magic for some user data
+  const did = magic.utils.parseAuthorizationHeader(req.headers.authorization);
+  const user = await magic.users.getMetadataByToken(did);
+
+  // Create a cookie to persist a user's session
+
+  res.end();
+};
+```
+
+<details open>
+  <summary><b>Examples</b></summary>
+  <ul>
+    <li><a href="https://github.com/vercel/next.js/tree/canary/examples/with-magic">with-magic</a></li>
+  </ul>
+</details>
+
+### Auth0
+
+Auth0 can support both authentication patterns. You can also utilize [API routes](https://nextjs.org/docs/api-routes/introduction) for logging in/out and retrieving user information. For example, after initializing the [Auth0 SDK](@auth0/nextjs-auth0), you can log-in by calling `handleLogin`.
+
+```jsx
+// pages/api/login.js
+
+import auth0 from '../../lib/auth0';
+
+export default async function login(req, res) {
+  try {
+    await auth0.handleLogin(req, res);
+  } catch (error) {
+    console.error(error);
+    res.status(error.status || 500).end(error.message);
+  }
+}
+```
+
+Auth0 populates the session so you can utilize static generation or `getServerSideProps` for server-side rendering. Here's an example of the latter.
+
+```jsx
+// pages/profile.js
+
+import auth0 from '../../lib/auth0';
+
+export async function getServerSideProps({ req, res }) {
+  const session = await auth0.getSession(req);
+
+  if (!session || !session.user) {
+    res.writeHead(307, { Location: '/api/login' });
+    res.end();
+    return { props: {} };
+  }
+
+  return { props: { user: session.user } };
+}
+```
+
+<details open>
+  <summary><b>Examples</b></summary>
+  <ul>
+    <li><a href="https://github.com/vercel/next.js/tree/canary/examples/auth0">auth0</a></li>
+  </ul>
+</details>
+
+## Frequently Asked Questions
+
+### How can I fetch user data once for every page in my application?
+
+To fetch data once for your entire Next.js application, create a [custom `App`](https://nextjs.org/docs/advanced-features/custom-app) and use `getInitialProps`.
+You should only use this method if you have blocking data requirements for _every single page_ in your application.
+This will disable [static optimization](https://nextjs.org/docs/advanced-features/automatic-static-optimization).
+We are [working on adding support](https://github.com/vercel/next.js/discussions/10949#discussioncomment-44898) for `getStaticProps` inside `_app.js`.
+
+Here's an example using a provider like Auth0.
+
+```jsx
+// _app.js
+
+App.getInitialProps = async ({ req, res }) => {
+  // Only run on the server
+  if (typeof window === 'undefined') {
+    const session = await auth0.getSession(req);
+    if (!session || !session.user) {
+      res.writeHead(307, { Location: '/api/login' });
+      res.end();
+
+      return;
+    }
+
+    return { user: session.user };
+  }
+};
+```
```

Comment:
Let's not add this. Highly unrecommended

Reply:
Ditch the entire section, or just the `getInitialProps` part?

Reply:
Ditch the entire section 👍

Reply:
Gone! 💀